### PR TITLE
Remove server version upgrade

### PR DESCRIPTION
**Pull request by Benjamin2037**
### First-time contributors' checklist 

- [X] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.
### What is changed, added or deleted? (Required)
增加关于在升级过程中，对于 server-version 配置参数的说明。

updates for TiDB in K8s is included in https://github.com/pingcap/docs-tidb-operator/pull/2431

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [X] master (the latest development version)
- [X] v7.3 (TiDB 7.3 versions)
- [X] v7.2 (TiDB 7.2 versions)
- [X] v7.1 (TiDB 7.1 versions)
- [X] v7.0 (TiDB 7.0 versions)
- [X] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -123,6 +123,9 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：""
 + 默认情况下，TiDB 版本号格式为：`5.7.${mysql_latest_minor_version}-TiDB-${tidb_version}`。
 
+> **注意：**
+> server-version 会被用于 TiDB 节点来验证当前 TiDB 的版本，因此在进行 TiDB 集群升级前，请将 server-version 清理掉，或者设置为当前 TiDB 真实的版本值，避免出现非预期行为。
```
Comment (ran-huang) [inline tidb-configuration-file.md:127]: 这个 note 也应该在[升级指南](https://docs.pingcap.com/zh/tidb/stable/upgrade-tidb-using-tiup#2-%E5%8D%87%E7%BA%A7%E5%89%8D%E5%87%86%E5%A4%87)里加上吧

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -123,6 +123,9 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：""
 + 默认情况下，TiDB 版本号格式为：`5.7.${mysql_latest_minor_version}-TiDB-${tidb_version}`。
 
+> **注意：**
```
Comment (ran-huang) [inline tidb-configuration-file.md:126]:
```suggestion
> **注意：**
>
```

File: tidb-configuration-file.md (hunk 1 above)
Comment (ran-huang) [inline tidb-configuration-file.md:127]:
```suggestion
> `server-version` 的值会被 TiDB 节点用于验证当前 TiDB 的版本。因此在进行 TiDB 集群升级前，请将 `server-version` 的值设置为空或者当前 TiDB 真实的版本值，避免出现非预期行为。
```

File: upgrade-tidb-using-tiup.md [hunk 3]
```diff
@@ -22,7 +22,8 @@ aliases: ['/docs-cn/dev/upgrade-tidb-using-tiup/','/docs-cn/dev/how-to/upgrade/u
 
 > **注意：**
 >
-> 如果原集群是 3.0 或 3.1 或更早的版本，不支持直接升级到 v7.2.0 及后续修订版本。你需要先从早期版本升级到 4.0 后，再从 4.0 升级到 v7.2.0 及后续修订版本。
+> 如果原集群是 3.0 或 3.1 或更早的版本，不支持直接升级到 v7.2.0 及后续修订版本。你需要先从早期版本升级到 4.0 后，再从 4.0 升级到 v7.2.0 及后续修订版本。 
+> 配置参数 ['server-version'](/tidb-configuration-file.md#server-version) 会被用于 TiDB 节点来验证当前 TiDB 的版本，因此在进行 TiDB 集群升级前，请将 server-version 清理掉，或者设置为当前 TiDB 真实的版本值，避免出现非预期行为。
```
Comment (ran-huang) [inline upgrade-tidb-using-tiup.md:26]:
```suggestion
> 配置参数 [`server-version`](/tidb-configuration-file.md#server-version) 的值会被 TiDB 节点用于验证当前 TiDB 的版本。因此在进行 TiDB 集群升级前，请将 `server-version` 的值设置为空或者当前 TiDB 真实的版本值，避免出现非预期行为。
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**